R/unnest-wider.R

```diff
@@ -0,0 +1,238 @@
+#' Unnest a list-column into columns
+#'
+#' @description
+#' `unnest_wider()` turns each element of a list-column into a column. It
+#' is most naturally suited to list-columns where every element is named,
+#' and the names are consistent from row-to-row.
```

Comment:
Having these "it is most naturally suited for..." in both `unnest_longer()` and `unnest_wider()` is really nice. The descriptions here make it pretty clear how to decide between the two